cmd/tk/util.go

```diff
@@ -35,22 +34,42 @@ func pageln(i ...interface{}) {
 
 	// invoke pager
 	cmd := exec.Command(pager, args...)
-	cmd.Stdin = strings.NewReader(fmt.Sprint(i...))
+	cmd.Stdin = r
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
 	// if this fails, just print it
 	if err := cmd.Run(); err != nil {
-		fmt.Print(i...)
+		io.Copy(os.Stdout, r)
 	}
 }
 
-func highlight(lang, s string) string {
-	var buf bytes.Buffer
-	if err := quick.Highlight(&buf, s, lang, "terminal", "vim"); err != nil {
-		log.Fatalln("Highlighting:", err)
+func colordiff(d string) io.Reader {
+	buf := bytes.Buffer{}
+
+	fmt.Fprintln(&buf)
+
+	lines := strings.Split(d, "\n")
+	for _, l := range lines {
+		switch {
+		case match(l, `^\+.*`):
+			color.New(color.FgGreen).Fprintln(&buf, l)
+		case match(l, `^-.*`):
+			color.New(color.FgRed).Fprintln(&buf, l)
+		case match(l, `^diff -u -N.*`):
+			color.New(color.FgBlue, color.Bold).Fprintln(&buf, l)
+		case match(l, `^@.*`):
+			color.New(color.FgMagenta, color.Bold).Fprintln(&buf, l)
+		default:
+			fmt.Fprintln(&buf, l)
+		}
 	}
-	return buf.String()
+
+	return &buf
+}
+
+func match(s, exp string) bool {
+	return regexp.MustCompile(exp).MatchString(s)
```

Comment:
nit: you could try to find a way to compile the list of expressions only once vs for each line.